azext_iot/tests/iothub/devices/test_hub_export_import.py

```diff
@@ -0,0 +1,364 @@
+        labels = json.dumps(labels)
+
+        priority = random.randint(1, 10)
+
+        metrics = {"metrics": {
+            "queries": {
+                "mymetric": "SELECT deviceId from devices where properties.reported.lastDesiredStatus.code = 200"
+            }
+        }}
+        metrics = json.dumps(metrics)
+
+        self.kwargs["config_content"] = config_content
+        self.kwargs["labels"] = labels
+        self.kwargs["metrics"] = metrics
+
+        self.cmd("iot hub configuration create --config-id {} -l {} --content '{}' --labels '{}' --priority {} --metrics '{}' \
+            --target-condition {}".format("hubConfig", self.connection_string, "{config_content}", "{labels}", priority,
+                                          "{metrics}", "tags.bar=12"))
+
+        # populate hub with devices
+
+        for device_type in DEVICE_TYPES:
+            device_count = 3
+            device_ids = self.generate_device_names(
+                device_count, edge=device_type == "edge"
+            )
+            edge_enabled = "--edge-enabled" if device_type == "edge" else ""
+
+            # Create SAS-auth device
+            custom_primary_key = generate_key()
+            custom_secondary_key = generate_key()
+            self.cmd(
+                f"iot hub device-identity create -d {device_ids[0]} -l {self.connection_string} --pk {custom_primary_key} \
+                    --sk {custom_secondary_key} {edge_enabled}"
+            )
+
+            # create x509_ca device
+            self.cmd(
+                f"iot hub device-identity create -d {device_ids[1]} -l {self.connection_string} --am x509_ca {edge_enabled}"
+            )
+
+            # create x509_thumbprint device
+            self.cmd(
+                f"iot hub device-identity create -d {device_ids[2]} -l {self.connection_string} --am x509_thumbprint \
+                    --ptp {PRIMARY_THUMBPRINT} --stp {SECONDARY_THUMBPRINT} {edge_enabled}"
+            )
+
+            # add a property and a tag to each device's twin
+            for i in range(len(device_ids)):
+                device = device_ids[i]
+
+                val = generate_generic_id()
+                self.cmd(
+                    f"iot hub device-twin update -d {device} -l {self.connection_string} --set properties.desired.testProp={val}"
+                )
+
+                patch_tags = {
+                    generate_generic_id(): generate_generic_id(),
+                    generate_generic_id(): generate_generic_id(),
+                }
+
+                self.kwargs["patch_tags"] = json.dumps(patch_tags)
+
+                self.cmd(
+                    f"iot hub device-twin update -d {device} -l {self.connection_string}"
+                    " --tags '{patch_tags}'"
+                )
+
+                # make a module for each device, same authentication method as device
+
+                if(i == 0):
```

Comment:
rather than putting this in the for loop (and awkwardly trying to force it to only occur on specific devices), why not put it right after you create the devices?